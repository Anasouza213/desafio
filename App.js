import React from 'react';
import { StyleSheet, View } from 'react-native';
import 'react-native-gesture-handler';
import { NavigationContainer } from '@react-navigation/native';
import Header from './src/components/header/Header'
import Routes from './src/routes';

const App = () => {
  return (
   <NavigationContainer >
     <View style={styles.container}>
            <Header>RESGATE</Header>
            <Routes />  
      </View>        
            
   </NavigationContainer>
  );
  
}
const styles = StyleSheet.create({
  container:{
    //marginTop: Platform.OS === 'ios' ? 20 : StatusBar.currentHeight,
    flex: 1
  }
})

export default App


