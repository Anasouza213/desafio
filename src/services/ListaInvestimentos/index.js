export const ListaInvestimentosService = () => {
    return fetch('https://run.mocky.io/v3/3c3cc3a1-151c-4937-8b4c-7694b524cc87')
    .then(resp => {
       return resp.json();
    })
  }
// import axios from 'axios';
// export const ListaInvestimentosService = () => {
// axios({
//    url: 'https://run.mocky.io/v3/3c3cc3a1-151c-4937-8b4c-7694b524cc87',
//    method: 'get',
//    data: {
//      query: `
//      query allPosts {
//        posts(listType: "all") {
//          text
//        }
//       }
//      `,
//    }
//  }).then(result => {
//    console.log(result.data);
//     return result.data;
   

//  }).catch(function (error) {
//    console.log(error);
//    console.log(error.request._response);
//  })
// }