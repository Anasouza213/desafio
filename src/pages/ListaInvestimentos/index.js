import React, { useEffect, useState } from 'react'
import { SafeAreaView, View, StyleSheet, Text } from 'react-native';
import {ListaInvestimentosService} from '../../services/index';
import ListaInvestimentosComponente from '../../components/listaInvestimentos/ListaInvestimentosComponente';

const ListaInvestimentos = () => {
    const [listaInvestimentos, setListaInvestimentos] = useState([]);


    useEffect( () => {
          async function getData(){
             await ListaInvestimentosService()
             .then(({  response  }) =>{
              setListaInvestimentos(response.data.listaInvestimentos);
             });
          }
          getData();
     });
   
    return (       
            <SafeAreaView style={styles.container}>
                <View style={{backgroundColor: '#f4f4f4', padding: 15}}>
                  <View style={{marginLeft: 20}}>
                    <Text >INVESTIMENTO</Text>
                    <Text style={{marginLeft: 210, marginTop: -19}}>R$</Text>
                  </View>                  
                </View>
                
                {listaInvestimentos ?
                  <ListaInvestimentosComponente lista={listaInvestimentos}/> :
                  <Text>Carregando Investimentos...</Text>
                  
                }
            </SafeAreaView>
    )
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
     
    },
    disabled:{
      padding: 5,      
      marginVertical: 0,    
      backgroundColor : '#fcfcfc',
    },
    item: {
      padding: 5,      
      marginVertical: 0,      
      backgroundColor: '#ffffff'     
    },
    border: {
      padding: 5,      
      borderBottomWidth: 1,
      borderBottomColor: "#ccc",
      marginTop: 20
    },
    textTitle: {
      fontSize: 15,
      color: "#34495e",
      maxWidth: 170
    },
    text: {
      fontSize: 13,
      color: "#34495e",
      maxWidth: 170
    },
    textValor: {
      fontSize: 15,
      color: "#34495e",
      marginLeft: 200,
      marginTop: -35
    },
    
  });

export default ListaInvestimentos;