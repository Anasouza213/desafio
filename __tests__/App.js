import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import ListaInvestimentos from '../src/pages/ListaInvestimentos';

test('App', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ListaInvestimentos', () => {
  const tree = renderer.create(<ListaInvestimentos />).toJSON();
  expect(tree).toMatchSnapshot();
});
