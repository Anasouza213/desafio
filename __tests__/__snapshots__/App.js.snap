// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignContent": "space-around",
          "alignItems": "center",
          "backgroundColor": "#005aa5",
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlignVertical": "top",
          }
        }
      >
        RESGATE
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fae128",
          "padding": 3,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              needsOffscreenAlphaCompositing={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fdfdfd",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        Object {
                          "backgroundColor": "#fdfdfd",
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "#f4f4f4",
                              "padding": 15,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginLeft": 20,
                              }
                            }
                          >
                            <Text>
                              INVESTIMENTO
                            </Text>
                            <Text
                              style={
                                Object {
                                  "marginLeft": 210,
                                  "marginTop": -19,
                                }
                              }
                            >
                              R$
                            </Text>
                          </View>
                        </View>
                        <View>
                          <RCTScrollView
                            data={Array []}
                            disableVirtualization={false}
                            getItem={[Function]}
                            getItemCount={[Function]}
                            horizontal={false}
                            initialNumToRender={10}
                            keyExtractor={[Function]}
                            maxToRenderPerBatch={10}
                            onContentSizeChange={[Function]}
                            onEndReachedThreshold={2}
                            onLayout={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            removeClippedSubviews={false}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={Array []}
                            updateCellsBatchingPeriod={50}
                            viewabilityConfigCallbackPairs={Array []}
                            windowSize={21}
                          >
                            <View />
                          </RCTScrollView>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 64,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </RNCSafeAreaProvider>
  </View>
</View>
`;
